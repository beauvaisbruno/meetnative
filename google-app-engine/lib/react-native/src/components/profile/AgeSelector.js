"use strict";
var __importDefault =
  (this && this.__importDefault) ||
  function(mod) {
    return mod && mod.__esModule ? mod : { default: mod };
  };
var __importStar =
  (this && this.__importStar) ||
  function(mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null)
      for (var k in mod)
        if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
  };
Object.defineProperty(exports, "__esModule", { value: true });
const native_base_1 = require("native-base");
const lang_1 = __importDefault(require("./lang"));
const React = __importStar(require("react"));
const custom_1 = __importDefault(
  require("../../config/native-base-theme/variables/custom")
);
const createRows = () => {
  const rows = [];
  for (let i = 18; i < 77; i++) {
    rows.push(
      React.createElement(native_base_1.Picker.Item, {
        label: lang_1.default("{age} yo", { age: i }),
        value: i,
        key: i
      })
    );
  }
  return rows;
};
exports.AgeSelector = ({ value, onChange }) => {
  return React.createElement(
    native_base_1.Picker,
    {
      testID: "picker",
      mode: "dropdown",
      iosIcon: React.createElement(native_base_1.Icon, { name: "arrow-down" }),
      placeholder: lang_1.default("Select your age"),
      placeholderStyle: { color: custom_1.default.textGrey },
      placeholderIconColor: custom_1.default.textGrey,
      style: { width: undefined },
      selectedValue: value,
      onValueChange: age => {
        onChange(age);
      }
    },
    createRows()
  );
};
//# sourceMappingURL=AgeSelector.js.map
