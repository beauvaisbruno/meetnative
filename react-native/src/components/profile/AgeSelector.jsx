"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var native_base_1 = require("native-base");
var lang_1 = require("./lang");
var React = require("react");
var custom_1 = require("../../config/native-base-theme/variables/custom");
var createRows = function () {
    var rows = [];
    for (var i = 18; i < 77; i++) {
        rows.push(
        //@ts-ignore
        <native_base_1.Picker.Item label={lang_1.default("{age} yo", { age: i })} value={i} key={i}/>);
    }
    return rows;
};
exports.AgeSelector = function (_a) {
    var value = _a.value, onChange = _a.onChange;
    return (<native_base_1.Picker testID={"picker"} mode="dropdown" iosIcon={<native_base_1.Icon name="arrow-down"/>} placeholder={lang_1.default("Select your age")} placeholderStyle={{ color: custom_1.default.textGrey }} placeholderIconColor={custom_1.default.textGrey} style={{ width: undefined }} selectedValue={value} onValueChange={function (age) {
        onChange(age);
    }}>
      {createRows()}
    </native_base_1.Picker>);
};
