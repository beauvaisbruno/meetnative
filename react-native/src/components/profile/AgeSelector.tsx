import { Icon, Picker } from "native-base";
import lang from "./lang";
import * as React from "react";
import custom from "../../config/native-base-theme/variables/custom";

const createRows = () => {
  const rows = [];
  for (let i = 18; i < 77; i++) {
    rows.push(
      //@ts-ignore
      <Picker.Item label={lang("{age} yo", { age: i })} value={i} key={i} />
    );
  }
  return rows;
};

export const AgeSelector = ({
  value,
  onChange
}: {
  value: number;
  onChange: (age: number) => void;
}) => {
  return (
    <Picker
      testID={"picker"}
      mode="dropdown"
      iosIcon={<Icon name="arrow-down" />}
      placeholder={lang("Select your age")}
      placeholderStyle={{ color: custom.textGrey }}
      placeholderIconColor={custom.textGrey}
      style={{ width: undefined }}
      selectedValue={value}
      onValueChange={(age: number) => {
        onChange(age);
      }}
    >
      {createRows()}
    </Picker>
  );
};
