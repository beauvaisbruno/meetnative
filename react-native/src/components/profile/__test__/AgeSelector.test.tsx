import { shallow } from "enzyme";
import { AgeSelector } from "../AgeSelector";
import * as React from "react";
import toJson from "enzyme-to-json";

it("Age Selector", async () => {
  const onChange = jest.fn();
  const tree = shallow(<AgeSelector value={1} onChange={onChange} />);
  expect(toJson(tree)).toMatchSnapshot();
  const picker = tree.findWhere(node => node.prop("testID") === "picker");
  picker.props().onValueChange(10);
  expect(onChange).toBeCalledWith(10);
});
