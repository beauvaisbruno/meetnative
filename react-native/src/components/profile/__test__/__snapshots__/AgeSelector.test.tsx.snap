// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Age Selector 1`] = `
<Styled(PickerNB)
  iosIcon={
    <Styled(Icon)
      name="arrow-down"
    />
  }
  mode="dropdown"
  onValueChange={[Function]}
  placeholder="Select your age"
  placeholderIconColor="#6d6d6d"
  placeholderStyle={
    Object {
      "color": "#6d6d6d",
    }
  }
  selectedValue={1}
  style={
    Object {
      "width": undefined,
    }
  }
  testID="picker"
>
  <Item
    key="18"
    label="18 yo"
    value={18}
  />
  <Item
    key="19"
    label="19 yo"
    value={19}
  />
  <Item
    key="20"
    label="20 yo"
    value={20}
  />
  <Item
    key="21"
    label="21 yo"
    value={21}
  />
  <Item
    key="22"
    label="22 yo"
    value={22}
  />
  <Item
    key="23"
    label="23 yo"
    value={23}
  />
  <Item
    key="24"
    label="24 yo"
    value={24}
  />
  <Item
    key="25"
    label="25 yo"
    value={25}
  />
  <Item
    key="26"
    label="26 yo"
    value={26}
  />
  <Item
    key="27"
    label="27 yo"
    value={27}
  />
  <Item
    key="28"
    label="28 yo"
    value={28}
  />
  <Item
    key="29"
    label="29 yo"
    value={29}
  />
  <Item
    key="30"
    label="30 yo"
    value={30}
  />
  <Item
    key="31"
    label="31 yo"
    value={31}
  />
  <Item
    key="32"
    label="32 yo"
    value={32}
  />
  <Item
    key="33"
    label="33 yo"
    value={33}
  />
  <Item
    key="34"
    label="34 yo"
    value={34}
  />
  <Item
    key="35"
    label="35 yo"
    value={35}
  />
  <Item
    key="36"
    label="36 yo"
    value={36}
  />
  <Item
    key="37"
    label="37 yo"
    value={37}
  />
  <Item
    key="38"
    label="38 yo"
    value={38}
  />
  <Item
    key="39"
    label="39 yo"
    value={39}
  />
  <Item
    key="40"
    label="40 yo"
    value={40}
  />
  <Item
    key="41"
    label="41 yo"
    value={41}
  />
  <Item
    key="42"
    label="42 yo"
    value={42}
  />
  <Item
    key="43"
    label="43 yo"
    value={43}
  />
  <Item
    key="44"
    label="44 yo"
    value={44}
  />
  <Item
    key="45"
    label="45 yo"
    value={45}
  />
  <Item
    key="46"
    label="46 yo"
    value={46}
  />
  <Item
    key="47"
    label="47 yo"
    value={47}
  />
  <Item
    key="48"
    label="48 yo"
    value={48}
  />
  <Item
    key="49"
    label="49 yo"
    value={49}
  />
  <Item
    key="50"
    label="50 yo"
    value={50}
  />
  <Item
    key="51"
    label="51 yo"
    value={51}
  />
  <Item
    key="52"
    label="52 yo"
    value={52}
  />
  <Item
    key="53"
    label="53 yo"
    value={53}
  />
  <Item
    key="54"
    label="54 yo"
    value={54}
  />
  <Item
    key="55"
    label="55 yo"
    value={55}
  />
  <Item
    key="56"
    label="56 yo"
    value={56}
  />
  <Item
    key="57"
    label="57 yo"
    value={57}
  />
  <Item
    key="58"
    label="58 yo"
    value={58}
  />
  <Item
    key="59"
    label="59 yo"
    value={59}
  />
  <Item
    key="60"
    label="60 yo"
    value={60}
  />
  <Item
    key="61"
    label="61 yo"
    value={61}
  />
  <Item
    key="62"
    label="62 yo"
    value={62}
  />
  <Item
    key="63"
    label="63 yo"
    value={63}
  />
  <Item
    key="64"
    label="64 yo"
    value={64}
  />
  <Item
    key="65"
    label="65 yo"
    value={65}
  />
  <Item
    key="66"
    label="66 yo"
    value={66}
  />
  <Item
    key="67"
    label="67 yo"
    value={67}
  />
  <Item
    key="68"
    label="68 yo"
    value={68}
  />
  <Item
    key="69"
    label="69 yo"
    value={69}
  />
  <Item
    key="70"
    label="70 yo"
    value={70}
  />
  <Item
    key="71"
    label="71 yo"
    value={71}
  />
  <Item
    key="72"
    label="72 yo"
    value={72}
  />
  <Item
    key="73"
    label="73 yo"
    value={73}
  />
  <Item
    key="74"
    label="74 yo"
    value={74}
  />
  <Item
    key="75"
    label="75 yo"
    value={75}
  />
  <Item
    key="76"
    label="76 yo"
    value={76}
  />
</Styled(PickerNB)>
`;
